src/core/qgstemporalnavigationobject.cpp

```diff
@@ -333,8 +333,11 @@ QgsTemporalNavigationObject::AnimationState QgsTemporalNavigationObject::animati
 long QgsTemporalNavigationObject::findBestFrameNumberForFrameStart( const QDateTime &frameStart ) const
 {
   long bestFrame = 0;
-  QgsDateTimeRange testFrame = QgsDateTimeRange( frameStart, frameStart ); // creatng an 'instant' Range here
-  for ( long i = 0; i < totalFrameCount(); ++i )
+  QgsDateTimeRange testFrame = QgsDateTimeRange( frameStart, frameStart ); // creating an 'instant' Range here
+  // earlier we looped from frame 0 till totalFrameCount() here, but this loop grew gigantic if you switched to for example milliseconds
+  // that is why now we calculate a rough framestart
+  long long roughFrameStart = std::floor( ( frameStart - mTemporalExtents.begin() ).seconds() / mFrameDuration.seconds() );
+  for ( long i = roughFrameStart; i < totalFrameCount(); ++i )
```

Comment:
Won't we potentially need to check backwards in some circumstances here? Eg. if the initial guess was too large and the frame estimated actually sits after the desired frameStart, then we'd need to step backwards from the guess until the candidate frame start is <= frameStart...

But otherwise, good approach to fixing this!

Reply:
@nyalldawson my gut feeling is that this will never happen, because of the 'floor', AND the fact that you only use the begin of the old frame in combination with a timestep-frame with a length >0
BUT I'm not 100% sure... but in my experiments this all worked. 
I can add a comment or todo?